ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/DepositsValidator.java

```diff
@@ -64,10 +77,41 @@ public boolean validateDepositsRoot(final Block block) {
   class AllowedDeposits implements DepositsValidator {
 
     private static final Logger LOG = LoggerFactory.getLogger(AllowedDeposits.class);
+    private final Address depositContractAddress;
+
+    public AllowedDeposits(final Address depositContractAddress) {
+      this.depositContractAddress = depositContractAddress;
+    }
+
+    public Address getDepositContractAddress() {
+      return depositContractAddress;
+    }
 
     @Override
-    public boolean validateDeposits(final Optional<List<Deposit>> deposits) {
-      return true; // TODO 6110: Validate deposits in the next PR
+    public boolean validateDepositParameter(final Optional<List<Deposit>> deposits) {
+      return deposits.isPresent();
+    }
+
+    @Override
+    public boolean validateDeposits(final Block block, final List<TransactionReceipt> receipts) {
+      if (block.getBody().getDeposits().isEmpty()) {
+        LOG.warn("Deposits must not be null when Deposits are activated");
```

Comment:
```suggestion
        LOG.warn("Deposits must not be empty when Deposits are activated");
```

Reply:
small nitpick - null has a specific meaning in java and technically using an optional here means it is `empty` rather than `null`

Reply:
For context, I think this is following what I did from Withdrawals and I specifically used "null" in order to match the spec definition:
https://github.com/ethereum/execution-apis/blob/main/src/engine/shanghai.md#specification-3
The spec has been rewritten since then so it's not as obvious even for withdrawals https://github.com/ethereum/execution-apis/commit/939255f467f9c5700a65426159afdfc0e0a66060

and deposits doesn't even mention "null" anymore https://github.com/ethereum/execution-apis/blob/main/src/engine/experimental/eip6110.md

Reply:
yeah would be good to have it consistently make sense across the board - I did notice it's null in the json test files, but it didn't make sense just looking at the java code

Reply:
Cool, I will modify `DepositsValidator` and leave `WithdrawalsValidator` untouched